### ISSUE TITLE
Should PreControlT in pna.p4 be removed?

### ISSUE THREAD

**Issue by qobilidop** [p4lang/p4c#4099]
Since [PreControl block has been removed in PNA spec](https://github.com/p4lang/pna/pull/78). I wonder if PreControlT in pna.p4 (and related backends) should be removed too?

https://github.com/p4lang/p4c/blob/7f53a12719478890820e2024652cc5a1a7ae7a69/p4include/pna.p4#L673-L677

**Comment by jafingerhut** [comment]
I believe it should be removed.

That isn't the only change made to the pna.p4 include file since the versions in p4c/p4include were copied from the PNA repository, of course, and we should consider updating the other things, too, which might be more work than simply removing the precontrol block.

**Comment by jafingerhut** [comment]
@usha1830 A related question: Do you know whether both of the files p4include/pna.p4 and p4include/dpdk/pna.p4 are used for anything?  At least one of them is, I know, but if we removed p4include/pna.p4, would everything work just as well?

**Comment by usha1830** [comment]
> @usha1830 A related question: Do you know whether both of the files p4include/pna.p4 and p4include/dpdk/pna.p4 are used for anything? At least one of them is, I know, but if we removed p4include/pna.p4, would everything work just as well?

For running pna tests with p4test, p4include/pna.p4 is required.
p4include/dpdk/pna.p4 is required by p4c-dpdk. Some externs in dpdk have differences from the p4include/pna.p4. Hence this is also required.

**Comment by usha1830** [comment]
> I believe it should be removed.
> 
> That isn't the only change made to the pna.p4 include file since the versions in p4c/p4include were copied from the PNA repository, of course, and we should consider updating the other things, too, which might be more work than simply removing the precontrol block.

Agree. p4c-dpdk would need some code changes to adapt to the latest pna.p4
